### PR TITLE
feat: move map card to component

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

* Refactors the Map Card out to a stateless component `MapMemberCard`
* Adds a loading state which better the dimensions for the loaded state

## Screenshots/Videos

This first screen recording demonstrates the current behaviour. The card displays a small block containing "loading" before the user data is fetched and displayed. This results in the card being resized which feels a bit janky.

https://user-images.githubusercontent.com/472589/204007419-36eb0cfb-29e5-4e70-8ba6-ba79da3cb497.mov

The second screen recording demonstrates the proposed behaviour. Instead the card's loading state better matches the expected size _after_ the content has loaded. This creates a gentler transition from the loading to loaded state.

https://user-images.githubusercontent.com/472589/204007443-5126e881-20e5-45fc-9cb9-2a9cabc93e48.mov

### CODE REVIEW

**Comment by davehakkens** [comment]
Oe nice detail! Wasnt aware how clunky it was, but now I am.
So we cant go back.

Can we style the card a bit more? At some point that text became blue but shouldn't, never reverted.
- Main text `color: black`
- Last active `font-size: 11px`

map-styling

**Comment by thisislawatts** [comment]
@davehakkens updated styling. Latest version:

Screenshot 2022-11-27 at 15 22 57

File: src/pages/Maps/Content/View/Popup.tsx [hunk 1]
```diff
@@ -83,12 +83,16 @@ export class Popup extends React.Component<IProps> {
     return 'loading'
   }
 
-  private renderContent(pin: IMapPinWithDetail) {
+  private getHeading(pin): string {
```
Comment (asheerrizvi) [inline src/pages/Maps/Content/View/Popup.tsx:86]: Should we specify the type for pin here or is it implied that it will be `IMapPinWithDetail`?

**Comment by asheerrizvi** [comment]
LGTM

**Comment by asheerrizvi** [comment]
Oops missed that there was a bunch of unused code. :x

**Comment by thisislawatts** [comment]
No worries @asheerrizvi, humans shouldn't need to spot unused code :) We should probably we revisit the eslint configuration to better automate some of this validation. 

I've raised https://github.com/ONEARMY/community-platform/pull/2018 to error on the implicit any type issue you flagged. [277 violations](https://app.circleci.com/pipelines/github/ONEARMY/community-platform/3343/workflows/6bef9025-2641-4796-80c2-b1688b054f4d/jobs/20970) across the codebase at the moment. Perhaps worth raising in a maintainer call to see if it is something that folks want to adopt.

**Comment by thisislawatts** [comment]
Failed at build step, https://app.circleci.com/pipelines/github/ONEARMY/community-platform/3342/workflows/9e7d89cc-e4ed-40b2-b6b7-16b9504594e0/jobs/20969, could have been caught earlier. 

> Treating warnings as errors because process.env.CI = true.
> Most CI servers set it automatically.

**Comment by asheerrizvi** [comment]
> No worries @asheerrizvi, humans shouldn't need to spot unused code :) We need to revisit the eslint configuration and automate this validation.
> 
> I've raised #2018 to error on the implicit any type issue you flagged. [277 violations](https://app.circleci.com/pipelines/github/ONEARMY/community-platform/3343/workflows/6bef9025-2641-4796-80c2-b1688b054f4d/jobs/20970) across the codebase at the moment. Perhaps worth raising in a maintainer call to see if it is something that folks want to adopt.

Awesome, no implicit any should be the norm.

**Review by davehakkens — APPROVED** [review]

**Pull request merged**